File: python/gsc/conf/dns/rule.py
from typing import Optional

import pydantic


class Rule(pydantic.BaseModel):
    auth_user: Optional[list[str]] = None
    domain_suffix: Optional[list[str]] = None
    clash_mode: Optional[str] = None
    rule_set: Optional[list[str]] = None
    outbound: Optional[list[str]] = None
    server: str
    disable_cache: Optional[bool] = None

File: src/internal/singbox/conf/dns/rule.rs
use serde::{Deserialize, Serialize};

#[serde_with::skip_serializing_none]
#[derive(Debug, Clone, PartialEq, Serialize, Deserialize, Default)]
pub struct Rule {
    pub auth_user: Option<Vec<String>>,
    pub clash_mode: Option<String>,
    pub rule_set: Option<Vec<String>>,
    pub outbound: Option<Vec<String>>,
    pub server: String,
    pub disable_cache: Option<bool>,
}
